modules/test/playwright/tests/client-extension-web/pages/ClientExtensionsPage.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+import {ApplicationsMenuPage} from '../../../pages/product-navigation-applications-menu/ApplicationsMenuPage';
+
+export class ClientExtensionsPage {
+	readonly applicationsMenuPage: ApplicationsMenuPage;
+	readonly itemActionsToggleButton: Locator;
+	readonly itemDeleteButton: Locator;
+	readonly itemEditButton: Locator;
+	readonly newClientExtensionButton: Locator;
+	readonly editorConfigContributorMenuItem: Locator;
+	readonly page: Page;
+
+	constructor(page: Page) {
+		this.applicationsMenuPage = new ApplicationsMenuPage(page);
+		this.page = page;
+
+		this.newClientExtensionButton = page
+			.getByRole('button')
+			.and(page.getByTitle('New'));
+
+		this.editorConfigContributorMenuItem = page.getByRole('menuitem', {
+			name: 'Add Editor Config Contributor',
+		});
+
+		this.itemActionsToggleButton = page.getByRole('button', {
+			name: 'Actions',
+		});
+
+		this.itemDeleteButton = page.getByRole('menuitem', {
+			name: 'Delete',
+		});
+		this.itemEditButton = page.getByRole('menuitem', {
+			name: 'Edit',
+		});
```

Comment:
This is assuming that no other client extensions were created previous to this test. If there are other CXs, this will fail. The test creates a CX entry and deletes it, so it it repeatable, unless it fails.

There are two ways this can be improved, but I'm not sure they are worth doing:
1. We could find the exact CX created by the test. This complicates locator, and also needs to work with pagination.
2. We could unconditionally clean up all CXs, even if tests fail.

Reply:
Mmm, why don't you turn the fields into methods and parameterize them? :thinking: 

Eg: change `itemEditButton` to `getItemEditButton(cxName: string): Locator` so that the caller (which is who knows the CX to be used) can specify which row to click. Would it work?

There may be a problem with pagination but that would be "your problem" in the sense that it is guaranteed that when tests start the DB is clean so you can assume there will be no CX unless other tests in your project create them and don't delete them upon exit (that's why I said it's "your" problem, because it is circumscribed to your tests only)

Reply:
Yep, something like a parameterized may make sense, but it's not really necessary for this test. Later on, if we need to create multiple CXs for a test, we can do it. But for now, it's an unnecessary complication.